gradio/components/video.py

```diff
@@ -306,21 +325,40 @@ def _format_video(self, video: str | Path | None) -> FileData | None:
             and not processing_utils.video_is_playable(video)
         ):
             warnings.warn(
-                "Video does not have browser-compatible container or codec. Converting to mp4"
+                "Video does not have browser-compatible container or codec. Converting to mp4."
             )
             video = processing_utils.convert_video_to_playable_mp4(video)
         # Recalculate the format in case convert_video_to_playable_mp4 already made it the selected format
         returned_format = utils.get_extension_from_file_path_or_url(video).lower()
-        if self.format is not None and returned_format != self.format:
-            if wasm_utils.IS_WASM:
-                raise wasm_utils.WasmUnsupportedError(
-                    "Returning a video in a different format is not supported in the Wasm mode."
+
+        if self.watermark_file or (
+            self.format is not None and returned_format != self.format
+        ):
+            global_option_list = ["-y"]
+            output_file_name = video[0 : video.rindex(".") + 1]
+            if self.format is not None:
+                if returned_format != self.format and wasm_utils.IS_WASM:
+                    raise wasm_utils.WasmUnsupportedError(
+                        "Returning a video in a different format is not supported in the Wasm mode."
+                    )
```

Comment:
wasm doesn't support ffmpeg generally, so we can't add a watermark in ffmpeg, so I'd move this back to line 337 at the top of this if block. We can modify this message to say something like "Modifying a video is not supported in the Wasm mode"

Reply:
I guess you mean "we can't add a watermark in WASM"?
Addressed.